### PR TITLE
metrics: export a function to decide if a metric is builtin. 

### PR DESCRIPTION
**Pull request by mattyw**
Keeps the logic in one place

### CODE REVIEW

File: metrics.go [hunk 1]
```diff
@@ -17,13 +17,18 @@ import (
...
 const (
-	builtinMetricsPrefix = "juju"
+	BuiltinMetricPrefix = "juju"
 
 	// Supported metric types.
 	MetricTypeGauge    MetricType = "gauge"
 	MetricTypeAbsolute MetricType = "absolute"
 )
 
+// IsBuiltinMetric returns true if the metric key provided is in the builtin metric namespace.
```
Comment (frankban) [inline metrics.go]: Should we add a trivial test for this at this point?
[shrug]

File: metrics.go [hunk 2]
```diff
@@ -17,13 +17,18 @@ import (
 type MetricType string
 
 const (
-	builtinMetricsPrefix = "juju"
+	BuiltinMetricPrefix = "juju"
```
Comment (frankban) [inline metrics.go]: Do we need to export this as well? Sorry I miss the background for this change.

**Comment by frankban** [comment]
:+1: with a couple of minor questions. Thank you!

File: metrics.go (hunk 1 above)
Comment (rogpeppe) [inline metrics.go]: Standard phrasing would be something like:

    // IsBuiltinMetric reports whether the given metric key is in the
    // built-in metric name space.

("built-in" vs "builtin" and "name space" vs "namespace" are still
arguable, but "reports whether" is definitely better IMHO).

File: metrics.go (hunk 2 above)
Comment (rogpeppe) [inline metrics.go]: Does this actually need to be exported now?

File: metrics.go (hunk 1 above)
Comment (rogpeppe) [inline metrics.go]: But tbh I find it hard to get excited about using this rather than just exporting BuiltinMetricPrefix.

File: metrics.go (hunk 2 above)
Comment (alesstimec) [inline metrics.go]: Yes, it does, i think.. can used to compose the built-in metrics name in the core.

**Comment by rogpeppe** [comment]
LGTM but I think that only one thing should be exported - if you need BuiltinMetricPrefix, then axe the strings.HasPrefix wrapper.

File: metrics.go (hunk 1 above)
Comment (alesstimec) [inline metrics.go]: +1 to frankban: a simple test would be in place

**Comment by alesstimec** [comment]
LGTM

File: metrics.go [hunk 3]
```diff
@@ -70,7 +75,7 @@ func ReadMetrics(r io.Reader) (*Metrics, error) {
 		return &metrics, nil
 	}
 	for name, metric := range metrics.Metrics {
-		if strings.HasPrefix(name, builtinMetricsPrefix) {
+		if IsBuiltinMetric(name) {
```
Comment (tasdomas) [inline metrics.go]: This definitely needs a test.

**Comment by alesstimec** [comment]
LGTM

File: metrics.go [hunk 4]
```diff
@@ -17,13 +17,18 @@ import (
...
 const (
-	builtinMetricsPrefix = "juju"
+	builtinMetricPrefix = "juju"
 
 	// Supported metric types.
 	MetricTypeGauge    MetricType = "gauge"
 	MetricTypeAbsolute MetricType = "absolute"
 )
 
+// IsBuiltinMetric reports whether the given metric key is in the builtin metric namespace
```
Comment (frankban) [inline metrics.go]: Please complete the sentence.

**Pull request merged**